### ISSUE TITLE
PHP Notices and Warnings which are reported on wordpress.org

### ISSUE THREAD

**Issue by stodorovic** [Automattic/wp-super-cache#383]
There are various PHP Notices in debug.log:

https://wordpress.org/support/topic/php-errors-fill-log-on-cli-cron-requests/
https://wordpress.org/support/topic/bug-wp_cache_remove_index-causes-no-such-file-or-directory-error/

` PHP Warning: json_encode(): Invalid UTF-8 sequence in argument in ..../wp-super-cache/wp-cache-phase2.php on line 53`
Related: https://github.com/Automattic/wp-super-cache/issues/349

**Comment by stodorovic** [comment]
I've found this into debug.log:

```
[29-Aug-2017 09:27:45 UTC] PHP Stack trace:
[29-Aug-2017 09:27:45 UTC] PHP   1. {main}() ..../xmlrpc.php:0
[29-Aug-2017 09:27:45 UTC] PHP   2. wp_xmlrpc_server->serve_request() ..../xmlrpc.php:84
[29-Aug-2017 09:27:45 UTC] PHP   3. wp_xmlrpc_server->IXR_Server() ..../wp-includes/class-wp-xmlrpc-server.php:197
[29-Aug-2017 09:27:45 UTC] PHP   4. wp_xmlrpc_server->__construct() ..../wp-includes/IXR/class-IXR-server.php:35
[29-Aug-2017 09:27:45 UTC] PHP   5. wp_xmlrpc_server->serve() ..../wp-includes/IXR/class-IXR-server.php:27
[29-Aug-2017 09:27:45 UTC] PHP   6. wp_xmlrpc_server->call() ..../wp-includes/IXR/class-IXR-server.php:65
[29-Aug-2017 09:27:45 UTC] PHP   7. wp_xmlrpc_server->pingback_ping() ..../wp-includes/IXR/class-IXR-server.php:115
[29-Aug-2017 09:27:45 UTC] PHP   8. wp_new_comment() ..../wp-includes/class-wp-xmlrpc-server.php:6431
[29-Aug-2017 09:27:45 UTC] PHP   9. do_action() ..../wp-includes/comment.php:1981
[29-Aug-2017 09:27:45 UTC] PHP  10. WP_Hook->do_action() ..../wp-includes/plugin.php:453
[29-Aug-2017 09:27:45 UTC] PHP  11. WP_Hook->apply_filters() ..../wp-includes/class-wp-hook.php:323
[29-Aug-2017 09:27:45 UTC] PHP  12. Gr_Integration->GrabEmailFromComment() ..../wp-includes/class-wp-hook.php:300
[29-Aug-2017 09:27:45 UTC] PHP Notice:  Constant DONOTDELETECACHE already defined in ..../wp-content/plugins/wp-super-cache/wp-cache-phase2.php on line 1282
```

**Comment by stodorovic** [comment]
Based on my experience, I created function `wpsc_is_backend` which returns true is `is_admin() == true` or PHP script is wp-login.php, wp-cron.php, xmlrpc.php. It's helpful because I can prevent creating output buffer in this case.

Related to wp-cron.php if it's running via shell script, it's tricky because SERVER_NAME doesn't exist. In this case `$WPSC_HTTP_HOST` is empty string. I made code which gets `$uri` via preg_match which is little faster and more flexible solution, but there is still issue at [this line](https://github.com/Automattic/wp-super-cache/blob/master/wp-cache-phase1.php#L661).

So, good advice for shell script is:
```
# Set server and execution environment information
export REMOTE_ADDR="127.0.0.1"
export REQUEST_METHOD="GET"
export SERVER_PROTOCOL="HTTP/1.1"
export SERVER_PORT="80"
export HTTP_USER_AGENT="Wp-cron/ver 1.0 (php-cli; Linux)"
export REQUEST_URI=/wp-cron.php
export HTTP_HOST="www.example.com"
export SERVER_NAME="www.example.com"
# change default dir to WP directory
cd public_html
# low priority
nice /usr/bin/php wp-cron.php
```

**Comment by stodorovic** [comment]
Closed #383 via #475.
